src/SoapCore/SoapEndpointMiddleware.cs

```diff
@@ -79,8 +79,8 @@ public SoapEndpointMiddleware(ILogger<SoapEndpointMiddleware<T_MESSAGE>> logger,
 
 		public async Task Invoke(HttpContext httpContext, IServiceProvider serviceProvider)
 		{
-			var trailPathTuner = serviceProvider.GetService<TrailingServicePathTuner>();
-
+			var trailPathTuner = serviceProvider.GetService<ITrailingServicePathTuner>();

```

Comment:
There is a failing unit test connected to this change. In it's current form this would be a breaking change.
```suggestion
			var trailPathTuner = serviceProvider.GetService<ITrailingServicePathTuner>() ??serviceProvider.GetService<TrailingServicePathTuner>();

```
With this change I _guess_ that we won't break existing code

Reply:
Sorry I haven't run any tests but I would say if it requires non-null when it should be GetRequiredService... meantime I see checking for null later... 'TrailPathTuner?.ConvertPath(httpContext)'
It seems the unit tests are not correct. (And please note that for Unity DI it will always be non-null with your suggestion and the only way is to create a dummy tuner)

Reply:
Sorry, I didn't explain this well enough.

Current usage depends on people registering TrailingServicePathTuner. A call to `serviceProvider.GetService<ITrailingServicePathTuner>()` won't hit that registration.
The code won't crash (it handles null) but the TrailingServicePathTuner won't be invoked, meaning that the behavior of existing services will change (if it depends on the tuner, that is).

Reply:
With my change the result will be null if you haven't registered a tuner at all

Reply:
An alternative to this fix is to rename TrailingServicePathTuner. That way people using it will get a compile time error and will (hopefully) investigate enough to find the interface.
What do you think?

Reply:
1) I agree, this can be overridden in a derived class that can be used for registration to provide custom path behavior. That fixes all problems.
2) Some Di frameworks (my case) can instantiate even an unregistered class. So the result could be not obvious. For this case, the interface is more suitable.